.github/workflows/merge.yml

```diff
@@ -4,13 +4,15 @@ on:
   push:
     branches:
       - develop
+  pull_request:
+    branches:
+      - develop
 
 jobs:
   tests:
-    runs-on: ${{ matrix.os }}
+    runs-on: macos-latest
```

Comment:
following the ticket from JIRA we agreed to:
"Run notebooks+unit tests+mac tests on PR and only unit tests on push"

shouldn't this workflow run all the test then and there should be a separate workflow for the push to the feature branch?

Reply:
That was what I had before e77e94fcdbe2131f6cb549f4c3fc5abfe2af49e9
The problem is that on push and on pull-request triggers at the same time and we have duplicate tests.
I propose this solution to keep the chosen behavior 
(PR and develop workflow run together on push devevelop and pull-request on develop, but on push on other branches, only develop workflow runs)

Reply:
I didn't find a way to execute only the PR workflow on pull-request.
So if duplicate tests isn't bothering anyone, I can revert this.
What do you think?

Reply:
It can be acceptable in the beginning to have second send of tests running during PR, as long as we don't run all the test suites on every push (it would take too long to execute).

Reply:
IMO we should have every test that is needed in `pull-requests` otherwise the tests that are 'exclusive' only for push won't run on pull requests from forks. Like right now no tests were run on two prs from contributors.